pkg/operator/ceph/object/realm/controller.go

```diff
@@ -0,0 +1,240 @@
+		// This handles the case where the operator is not ready to accept Ceph command but the cluster exists
+		if !cephObjectStoreRealm.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Remove finalizer
+			err = opcontroller.RemoveFinalizer(r.client, cephObjectStoreRealm)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
+			}
+
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		return reconcileResponse, nil
+	}
+
+	// Set a finalizer so we can do cleanup before the object goes away
+	err = opcontroller.AddFinalizerIfNotPresent(r.client, cephObjectStoreRealm)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to add finalizer")
+	}
+
+	// DELETE: the CR was deleted
+	if !cephObjectStoreRealm.GetDeletionTimestamp().IsZero() {
+		logger.Debugf("deleting realm CR %q", cephObjectStoreRealm.Name)
+
+		// Remove finalizer
+		err = opcontroller.RemoveFinalizer(r.client, cephObjectStoreRealm)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
+		}
+
+		// Return and do not requeue. Successful deletion.
+		return reconcile.Result{}, nil
+	}
+
+	// validate the realm settings
+	err = ValidateRealmCR(cephObjectStoreRealm)
+	if err != nil {
+		cephObjectStoreRealm.Status.Phase = k8sutil.ReconcileFailedStatus
+		err := opcontroller.UpdateStatus(r.client, cephObjectStoreRealm)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to set status")
+		}
+		return reconcile.Result{}, errors.Wrapf(err, "invalid name CR %q spec", cephObjectStoreRealm.Name)
+	}
+
+	// Start object reconciliation, updating status for this
+	cephObjectStoreRealm.Status.Phase = k8sutil.ReconcilingStatus
+	err = opcontroller.UpdateStatus(r.client, cephObjectStoreRealm)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to set status")
+	}
+
+	// CREATE/UPDATE CEPH REALM
+	reconcileResponse, err = r.reconcileCephRealm(cephObjectStoreRealm)
+	if err != nil {
+		cephObjectStoreRealm.Status.Phase = k8sutil.ReconcileFailedStatus
+		errStatus := opcontroller.UpdateStatus(r.client, cephObjectStoreRealm)
+		if errStatus != nil {
+			logger.Errorf("failed to set status. %v", errStatus)
+		}
+		return reconcileResponse, err
+	}
+
+	// Set Ready status, we are done reconciling
+	cephObjectStoreRealm.Status.Phase = k8sutil.ReadyStatus
+	err = opcontroller.UpdateStatus(r.client, cephObjectStoreRealm)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to set status")
+	}
+
+	// Return and do not requeue
+	logger.Debug("done reconciling")
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileObjectStoreRealm) reconcileCephRealm(cephObjectStoreRealm *cephv1.CephObjectStoreRealm) (reconcile.Result, error) {
+	if err := createRootPool(r.context, cephObjectStoreRealm.Name, cephObjectStoreRealm.Namespace, cephObjectStoreRealm.Spec.RootPoolSize); err != nil {
```

Comment:
There is already a function to create pools. See `CreateReplicatedPoolForApp()`.